### PR TITLE
Feature/master/passengerbaseuri

### PR DESCRIPTION
**Pull request by aronymous**
(MODULES-2188) Add support for PassengerBaseURI in Apache

### CODE REVIEW

File: manifests/vhost.pp [hunk 1]
```diff
@@ -385,6 +386,13 @@
     }
   }
 
+  # Load mod_passenger if needed and not yet loaded
+  if $passenger_base_uris {
+    if ! defined(Class['apache::mod::passenger']) {
```
Comment (mhaskel) [inline manifests/vhost.pp]: You don't need to have this check, `include ::apache::mod::passenger` will just work.

File: manifests/vhost.pp [hunk 2]
```diff
@@ -655,6 +663,16 @@
   }
 
   # Template uses:
+  # - $passenger_base_uris
+  if $passenger_base_uris {
+    concat::fragment { "${name}-passenger_uris":
+      target  => "${priority_real}${filename}.conf",
+      order   => 150,
```
Comment (hunner) [inline manifests/vhost.pp]: This should be 155, as the fragment above is 150

**Comment by mhaskel** [comment]
@aronymous thanks for the contribution! This needs a rebase, squash, and tests before this can be merged.

**Comment by aronymous** [comment]
I added tests and made the changes suggested. Hope I did it right. Thanks

**Comment by mhaskel** [comment]
Hi @aronymous ,

This still needs a squash, there is some pretty good documentation [here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request)

File: manifests/vhost.pp [hunk 3]
```diff
@@ -386,6 +387,11 @@
     }
   }
 
+  # Load mod_passenger if needed and not yet loaded
+  if $passenger_base_uris {
+      include ::apache::mod::passenger
```
Comment (hunner) [inline manifests/vhost.pp]: 4 spaces should be 2 space indentation.

**Comment by mhaskel** [comment]
Thanks @aronymous

**Pull request merged**